### ISSUE TITLE
Timezone information is ignored in google calendar event actions

### ISSUE THREAD

**Issue by data-overflow** [ComposioHQ/composio#1106]
### Summary
Composio requires the datetime to be in ISO format (which can optionally have timezone information) but assumes all datetime to be in UTC and adjusts the time before calling the functions. This behavior is undesired and causes problems while working with conversational agents.

### Example
The following two variants of code results in the same outcome 
**Case 1:**
```py
toolset.execute_action(
    action=Action.GOOGLECALENDAR_CREATE_EVENT,
    params={"start_datetime": "2024-12-30T07:30:00", "event_duration_hour": 2, "event_duration_minutes": 0, "summary": "test"},
)
```
**Case 2:**
```py
toolset.execute_action(
    action=Action.GOOGLECALENDAR_CREATE_EVENT,
    params={"start_datetime": "2024-12-30T07:30:00+05:30", "event_duration_hour": 2, "event_duration_minutes": 0, "summary": "test"},
)
```
**Output:**
![image](https://github.com/user-attachments/assets/f5a4f2a3-d8cc-42bb-a5b9-3832e3869588)

In both the above cases +5:30 hours is applied to match my current timezone although in the second case we explicitly mention the timezone. **Additionally, providing the "timezone" parameter in the params also has no effect**

**Comment by plxity** [comment]
@data-overflow Thanks for reporting this, I'm checking this issue

**Comment by data-overflow** [comment]
Yes, I appreciate it! The issue is reproducible from the dashboard as well.
![WhatsApp Image 2024-12-30 at 17 09 26_11ef263c](https://github.com/user-attachments/assets/633a365c-dd84-40ed-8f11-fd8438f85ebd)
**Here are the parameters I used:**
start_datetime: 2024-12-30T12:00:00+05:30 (12pm IST)
event_duration_hour: 2
event_duration_minutes: 1 (minor bug, the UI assumes 0 as null and throws "this field is required" error)
summary: test
timezone: Asia/Kolkata

**Output:**
![image](https://github.com/user-attachments/assets/32cfcb74-f704-4f96-9fd5-d4d9d1c2a045)

**Comment by Devanshusisodiya** [comment]
@data-overflow the datetime format `2024-12-30T07:30:00+05:30` is not supported by Google Calendar itself, although there is support for timezone by using the `Timezone` parameter. Just need to make sure the timezone being passed is valid.

<img width="601" alt="image" src="https://github.com/user-attachments/assets/a5f0080c-952e-466f-ac95-0b1e332bd578" />


Thanks. Let us know if there are any issues.

**Comment by data-overflow** [comment]
Hi @Devanshusisodiya, thank you for the response. I've indeed tested by passing the timezone in the required IANA format. Here's an example:
I used start datetime as: 2025-01-02T12:00:00 (12pm, without timezone information) 
![image](https://github.com/user-attachments/assets/c8f87d68-8f08-46c4-8be9-4c734959ce92)

The output is still processed to Asia/Kolkata time.
![image](https://github.com/user-attachments/assets/563dafb7-8616-403a-a08b-359fa4c62f9b)

Does the timezone require a completely different format than the one specified? Or is it by chance ignored entirely?

**Comment by Devanshusisodiya** [comment]
@data-overflow, this issue has been fixed. Closing this issue.
